components/script/dom/readablestream.rs

```diff
@@ -742,6 +740,57 @@ pub(crate) enum ReaderType {
     Default(MutNullableDom<ReadableStreamDefaultReader>),
 }
 
+impl Clone for ReaderType {
+    #[cfg_attr(crown, allow(crown::unrooted_must_root))]
+    fn clone(&self) -> Self {
+        match self {
+            ReaderType::BYOB(reader) => ReaderType::BYOB(MutNullableDom::new(Some(
+                &reader.get().expect("Reader should be set"),
+            ))),
+            ReaderType::Default(reader) => ReaderType::Default(MutNullableDom::new(Some(
+                &reader.get().expect("Reader should be set"),
+            ))),
+        }
+    }
+}
+
+impl ReaderType {
+    /// Get the stream from the reader.
+    pub(crate) fn get_stream(&self) -> Option<DomRoot<ReadableStream>> {
+        match self {
+            ReaderType::BYOB(reader) => reader.get().expect("Reader should be set").get_stream(),
+            ReaderType::Default(reader) => reader.get().expect("Reader should be set").get_stream(),
+        }
+    }
+
+    /// Release the reader.
+    pub(crate) fn release(&self, can_gc: CanGc) -> Fallible<()> {
+        match self {
+            ReaderType::BYOB(reader) => reader.get().expect("Reader should be set").release(can_gc),
+            ReaderType::Default(reader) => {
+                reader.get().expect("Reader should be set").release(can_gc)
+            },
+        }
+    }
+
+    // get reader Closed promise
+    pub(crate) fn closed(&self) -> Rc<Promise> {
+        match self {
+            ReaderType::BYOB(reader) => reader.get().expect("Reader should be set").Closed(),
+            ReaderType::Default(reader) => reader.get().expect("Reader should be set").Closed(),
+        }
+    }
+
+    // Read
+    pub(crate) fn read(&self, can_gc: CanGc) -> Rc<Promise> {
+        match self {
+            ReaderType::BYOB(_) => {
+                unreachable!("Native reading of a chunk can only be done with a default reader.")
```

Comment:
This doesn't seem right.

Reply:
What I mean is that here, unlike at `read_a_chunk`, the point is to actually use a `BYOB` reader for the native reading of chunk as part of the piping flow. And doing so will hit this branch of logic, and panic.  Is my understanding correct?